activestorage/test/models/preview_test.rb

```diff
@@ -70,6 +70,61 @@ class ActiveStorage::PreviewTest < ActiveSupport::TestCase
     assert_changes -> { !!preview.image.variant(transformations)&.send(:processed?) }, to: true do
       preview.processed
     end
+
+    image = read_image(preview.image.variant(transformations))
+    assert_equal 216, image.width
+    assert_equal 280, image.height
+  end
+
+  test "#processed does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed
+    end
```

Comment:
Instead of `assert_no_changes`, let's just assert that there are no variant records after all is done:

```suggestion
    preview.processed
    # etc

    assert_empty preview.image.variant_records
```